src/components/SideBar.js

```diff
@@ -91,8 +97,30 @@ Menu.propTypes = {
   title: PropTypes.string,
 };
 
+const fetchAnnouncements = async () => {
+  const res = await fetch(process.env.NEXT_PUBLIC_NEW_ANNOUNCEMENTS_URL);
+  const { data } = await res.json();
+  return data;
+};
+
 const SideBar = () => {
   const { t } = useTranslation();
+  const [isModalVisible, setModalVisibility] = useState(false);
+  const [announcements, setAnnouncements] = useState([]);
+  const [seenAnnouncements, setSeenAnnouncements] = useState([]);
+  const [countUnseenAnnouncements, setCountUnseenAnnouncements] = useState(0);
+
+  useEffect(async () => {
+    const newAnnouncements = await fetchAnnouncements();
+    setAnnouncements(newAnnouncements);
+  }, []);
+
+  useEffect(() => {
+    if (announcements.length === 0) return;
+    setCountUnseenAnnouncements(
+      announcements.length - seenAnnouncements.length,
+    );
+  }, [announcements, seenAnnouncements]);
```

Comment:
Can't we derive this instead of recalculating and storing it? If it's super intensive (which it isn't) we can wrap it in a useMemo.

Reply:
I was thinking we could maybe combine the `AnnouncementsModal` and `Inline` that serves as a link in a single component that fetches the announcements and keeps track of which are seen? And then the `AnnouncementsModal` is a stateless component that takes the data as props and has some kind of `onRead` prop so the parent component can update the state and also push that read count down to the badge?

Reply:
Discussed a different approach with @simon-debruijn in a call, we should be able to treat announcements as the single source of truth and derive everything from there

Reply:
Awesome 👍